vendor_data.py

```diff
@@ -56,101 +52,190 @@ def __init__(self, VendorName, TargLang, WordRate, VendorMail = "", CatTool = "X
             CatTool should be a string, if not a TypeError is raised.
             CatTool should be one of the options in the list "CatTools", if not a ValueError is raised.
         """
+        regex_mail = "^[a-z0-9]+[\._]?[a-z0-9]+[@]\w+[.]\w{2,3}$" #regex used to validate email
+        
         
+        if type(ProjectName) != str:
+            raise TypeError("ProjectName should be a string!")
+        else:
+            self.ProjectName = ProjectName
+        if type(SourceLang) != str:
+            raise TypeError("SourceLang should be a string!")
+        else:
+            self.SourceLang = SourceLang
         if type(VendorName) !=str:
             raise TypeError("VendorName should be a string!")
+        else:
+            self.VendorName = VendorName
+            
         if type(TargLang) != str:
             raise TypeError("TargLang should be a string!")
-        if type(WordRate) != float:
-            raise TypeError("WordRate should be a float!")
+        else:
+            self.TargLang = TargLang
+            
         if WordRate:
+            if type(WordRate) != float:
+                raise TypeError("WordRate should be a float!")
             if WordRate > 0.15:
-                raise ValueError("This vendor is too expensive, consider another one.")
-        elif WordRate == 0.00:
-            raise ValueError("Word rate cannot be 0.00.")
+                raise ValueError("This vendor is too expensive, pick another one.")
+            if WordRate == 0.00:
+                raise ValueError("Word rate cannot be 0.00.")
         else:
             self.WordRate = WordRate
-        if type(CatTool) != str:
-            raise TypeError("CatTool should be a string!")
-        if type(VendorMail) != str:
-            raise TypeError ("VendorMail should be a string!")
-        if VendorMail:   
+            
+        if Preferred != None:
+            if Preferred == True:
+                self.Status = "Preferred"
+            elif Preferred == False:
+                self.Status = "Back-up"
+        else:

```

Comment:
This is good, but just as a suggestion: lines 87-90 can be collapsed into `self.Status = "Preferred" if Preferred else "Back-up"`. It's the same thing. (`if Preferred` is the same as `if Preferred == True`)
Of course, this doesn't consider the possibility that `Status` is not boolean or `None`. If someone provides `Status = "Preferred"`, your solution will ignore the value whereas mine sets `"Preferred"`. And if someone provides `Status = "Back-up"`, your solution will ignore the value whereas mine sets `"Preferred"`. (Do you understand why?) The question is whether you want to inform the user if they submit an invalid value...